spring-cloud-gcp-samples/spring-cloud-gcp-secretmanager-sample/README.adoc

```diff
@@ -0,0 +1,54 @@
+= Spring Cloud GCP Secret Manager Sample Application
+
+This code sample demonstrates how to use the Spring Cloud GCP Secret Manager integration.
+
+To begin using Secret Manager support, first add the `spring-cloud-gcp-starter-secretmanager` dependency to the project:
+
+```
+<dependency>
+	<groupId>org.springframework.cloud</groupId>
+	<artifactId>spring-cloud-gcp-starter-secretmanager</artifactId>
+</dependency>
+```
+
+Adding the starter dependency will automatically cause your GCP Project's secrets to be injected into your application's environment at bootstrap time.
+
+This sample application defines a `@ConfigurationProperties` in order to access and use the secret values.
+
+```
+@ConfigurationProperties
+public class MyAppProperties {
+
+	@Value("${application-secret}")
+	private String applicationSecret;
+	...
+}
+```
+
+== Running the Sample
+
+1. Create a Google Cloud Platform project with https://cloud.google.com/billing/docs/how-to/modify-project#enable-billing[billing enabled], if you don't have one already.
+
+2. Enable the Secret Manager API from the "APIs & Services" menu of the Google Cloud Console.
+
+3. Authenticate in one of two ways:
+
+a. Use the Google Cloud SDK to https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login[authenticate with application default credentials].
+This method is convenient but should only be used in local development.
+b. https://cloud.google.com/iam/docs/creating-managing-service-accounts[Create a new service account], download its private key and point the `spring.cloud.gcp.secretmanager.credentials.location` property to it.
++
+Such as: `spring.cloud.gcp.secretmanager.credentials.location=file:/path/to/creds.json`
+
+4. Using the https://console.cloud.google.com/security/secret-manager[Secret Manager UI in Cloud Console], create a new secret named `application-secret` and set it to any value.
```

Comment:
Maybe do something like `secrets.password`, which will map to `MyAppProperties.password`.

Reply:
It turns out that secret names stored in Secret Manager can only contain English letters (A-Z), numbers (0-9), dashes (-), and underscores (_); so storing `secrets.password` is not possible.

Reply:
I made the sample demonstrate the namespace feature using `secrets.`